### PR TITLE
LPS-88714 Document API client tests

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

**Comment by ruben-pulido** [comment]
ci:test:sf <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ruben-pulido** [comment]
ci:test:relevant <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+		}
+
+		if (_folderHref != null) {
+			_delete(_folderHref);
+		}
+	}
+
+	@Test
+	public void testCreateDocumentInDocumentRepository() {
+		String documentsHref = ApioClientBuilder.given(
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java]: Could extract documentsHref as a setUp vairable?

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java [hunk 2]
```diff
@@ -0,0 +1,641 @@
...
+	private void _delete(String url) {
+		ApioClientBuilder.given(
+		).basicAuth(
+			"test@liferay.com", "test"
+		).when(
+		).delete(
+			url
+		).then(
+		).statusCode(
+			Matchers.isOneOf(200, 204)
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java]: Just delete, we don't need to assert anything, there is a test for that

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+		).header(
+			"Content-Type", "multipart/form-data"
+		).multipart(
+			"binaryFile", _readFile("document-updated.pdf")
+		).when(
+		).put(
+			_documentHref
+		).then(
+		).statusCode(
+			404
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java]: Don't test 404 error in functional test, test it as integration tests

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java (hunk 2 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java]: 👍

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Ruben Pulido
+ */
+@RunAsClient
+@RunWith(Arquillian.class)
+public class MediaOjectApioTest {
+
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java]: In general, simplify test following the same patters we already have in StructuredContentApioTest and BlogsApioTest.

Extract test initialization to methods ...

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java (hunk 1 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java]: 👍

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java (hunk 3 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java]: 👍

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java (hunk 4 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaOjectApioTest.java]: 👍

**Comment by ruben-pulido** [comment]
Thanks for all the feedback @cgoncas! I have incorporated [here](https://github.com/cgoncas/liferay-portal/pull/338)

**Pull request closed without merging**